src/MongoDB.Driver.Core/Core/Operations/ChangeStreamCursor.cs

```diff
@@ -137,10 +137,8 @@ private bool CanResumeAfter(Exception exception)
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times")]
         private TDocument DeserializeDocument(RawBsonDocument rawDocument)
         {
-            var slice = rawDocument.Slice;
-            var bytes = slice.AccessBackingBytes(0);
-            using (var memoryStream = new MemoryStream(bytes.Array, bytes.Offset, bytes.Count))
```

Comment:
The bug is that this MemoryStream only covers the bytes in *one* chunk, so if the rawDocument spilled over to the next chunk the MemoryStream was missing those bytes.